JS/Lesson34/goksel/recipe-app/src/modules/user/RegistrationForm/index.tsx

```diff
@@ -0,0 +1,170 @@
+import { useForm } from "react-hook-form";
+import { Link, unstable_HistoryRouter } from 'react-router-dom';
+import Avatar from '@mui/material/Avatar';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import { Box, Container, Grid, TextField, Typography, FormControl, InputLabel, OutlinedInput, InputAdornment, IconButton, FormControlLabel, Button, Snackbar, FormHelperText } from "@mui/material";
+import { useState } from "react";
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
+import Checkbox from '@mui/material/Checkbox';
+
+interface RegistrationFormProps {
+	open: boolean;
+	setOpen(open: boolean): void;
+	firstName: string;
+	lastName: string;
+	email: string;
+	password: string;
+}
+
+export const RegistrationForm = () => {
```

Comment:
Once a user registers, you need to add his/her data to the redux store by calling dispatch method for a certain action type (e.g. update). You'd need to add a new method (update) to userSlice to support this.